core/trino-main/src/main/java/io/trino/operator/output/RleAwarePositionsAppender.java

```diff
@@ -90,6 +90,13 @@ else if (rleValue != null) {
         }
     }
 
+    @Override
+    public void append(int position, Block value)
+    {
+        switchToFlat();
```

Comment:
we could handle rle here at some CPU cost but we don't handle it in the blockBuilder case anyway

Reply:
Yes, but appenders are not block builders nor they should behave like ones. Appenders are optimized to process big batches of positions (including special block types like RLEs and dictionaries). Here we quickly fallback to block-builder behaviour just because we've seen single row append.